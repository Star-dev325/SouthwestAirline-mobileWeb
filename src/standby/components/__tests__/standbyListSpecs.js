import { shallow } from 'enzyme';
import React from 'react';
import StandbyList from 'src/standby/components/standbyList';

describe('StandbyList', () => {
  let StandbyListComponent;

  beforeEach(() => {
    const standbyListProps = {
      standbyList: [
        {
          isConfirmed: true,
          displayName: 'OTHER / TI'
        },
        {
          isConfirmed: false,
          displayName: 'ONVAC / SI',
          number: '1'
        }
      ]
    };

    StandbyListComponent = shallow(<StandbyList {...standbyListProps} />);
  });

  it('should render standby list', () => {
    expect(StandbyListComponent.find('.confirmed')).to.have.length(1);
    expect(StandbyListComponent).to.contain.text('OTHER / TI');

    expect(StandbyListComponent.find('.unconfirmed')).to.have.length(1);
    expect(StandbyListComponent).to.contain.text('ONVAC / SI');
  });

  it('should render standby list enhanced view', () => {
    const standbyListProps = {
      isEnhancedStandby: true,
      standbyList: [
        {
          isConfirmed: true,
          displayName: 'OTHER / TI',
          isPnrPassenger: true
        },
        {
          isConfirmed: false,
          displayName: 'ONVAC / SI',
          number: '1'
        }
      ]
    };

    StandbyListComponent = shallow(<StandbyList {...standbyListProps} />);

    expect(StandbyListComponent).toMatchSnapshot();
  });
});
