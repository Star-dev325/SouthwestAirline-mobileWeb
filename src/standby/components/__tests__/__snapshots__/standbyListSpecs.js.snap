// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandbyList should render standby list enhanced view 1`] = `
<div
  className="standby-body--standby-list"
>
  <div
    className="standby-list--item enhanced-standby-list--item"
    key="0"
  >
    <div
      className="standby-list--item-left"
    >
      <div
        className="standby-list--item-circle confirmed"
      >
        <Icon
          className="confirm-check"
          type="check"
        />
      </div>
    </div>
    <div
      className="standby-list--item-right"
    >
      <span
        className="standby-list--item-name standby-list--item-name-pnr"
      >
        OTHER / TI
      </span>
      <span
        className="standby-list--item-cleared"
      >
        CLEARED
      </span>
    </div>
  </div>
  <div
    className="standby-list--item enhanced-standby-list--item"
    key="1"
  >
    <div
      className="standby-list--item-left"
    >
      <div
        className="standby-list--item-circle unconfirmed"
      >
        <span
          className="standby-list--item-number"
        >
          1
        </span>
      </div>
    </div>
    <div
      className="standby-list--item-right"
    >
      <span
        className="standby-list--item-name"
      >
        ONVAC / SI
      </span>
    </div>
  </div>
</div>
`;
